solr/core/src/test/org/apache/solr/core/TestCoreContainer.java

```diff
@@ -1131,6 +1132,48 @@ public void testCoreInitFailuresOnReload() throws Exception {
     cc.shutdown();
   }
 
+  public void testCustomSocketTimeoutForDefaultHttpClient() throws Exception {
```

Comment:
hmm; why this test here?

Reply:
I want to test where we can change the SocketTimeout via UpdateShardHandlerCfg, Now `TestCoreContainer` seems to have that code where we can initlialize the CoreContainer with custom SolrXML file. So I added those.

Reply:
Additionally, I've noticed that the documentation does not cover how to pass UpdateShardHandlerConfig or how it affects the timeouts for the HTTP/2 client—specifically, the update, recovery, and default timeouts.

Reply:
https://solr.apache.org/guide/solr/latest/configuration-guide/configuring-solr-xml.html

Reply:
I could see _learning_ from TestCoreContainer but putting your test there just to do something that could be done elsewhere doesn't seem like a good reason to put the test there.  Granted I'm also not sure where a good place for this is either :-).  So maybe here.

Must this thing be configured in solr.xml; can't we rely simply on EnvUtils ?  I have a wish-list JIRA issue to unify things in a more automatic way.  But in the mean time, I recommend preferring EnvUtils over solr.xml for simple (not structural / flexible) configuration.

Reply:
We can also just remove this test. The mock that I written is almost same what I am trying to test here via integration. I think I don't really need integration test in this case.